### ISSUE TITLE
I got 401 error

### ISSUE THREAD

**Issue by ghost** [becem-gharbi/nuxt-directus#91]
Hi, I got:
GET users/me?fields=* 401 (Unauthorized)
after login!!!

**Comment by mahsamirzapour** [comment]
Oh sorry my bad! I used directus sdk and this modules together and the issue caused! this module saved my life thank you

**Comment by becem-gharbi** [comment]
Hi, can you provide more info on this issue? a reproduction would be helpful.

**Comment by ghost** [comment]
> Hi, can you provide more info on this issue? a reproduction would be helpful.

Hi, first of all the module has an issue with node v22 lts, and I have to installed by --force flag, the users/me endpoint does not keep cookie or session and I got 401, the module has some issue with cookies and SSR

**Comment by becem-gharbi** [comment]
Have you installed the latest version of the module? If yes, feel free to provide a reproduction to help understand the issue, https://stackblitz.com/github/nuxt/starter/tree/v3.

**Comment by ghost** [comment]
@becem-gharbi  Yes but with --force flag, I installed the latest version, please test it with node v22 and SSR mode with Directus docker to understand the issue, I can't help for reproduction because I'm little bit newbie lol

**Comment by ghost** [comment]
@becem-gharbi  this is how I got on installation:

npm install --save-dev @bg-dev/nuxt-directus
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: nuxt-app@undefined
npm error Found: vue@3.5.13
npm error node_modules/vue
npm error   peer vue@"^3.3.4" from @nuxt/vite-builder@3.15.0
npm error   node_modules/@nuxt/vite-builder
npm error     @nuxt/vite-builder@"3.15.0" from nuxt@3.15.0
npm error     node_modules/nuxt
npm error       nuxt@"^3.15.0" from the root project
npm error   peer vue@">=2.7 || >=3" from @unhead/vue@1.11.14
npm error   node_modules/@unhead/vue
npm error     @unhead/vue@"^1.11.14" from nuxt@3.15.0
npm error     node_modules/nuxt
npm error       nuxt@"^3.15.0" from the root project
npm error   8 more (@vitejs/plugin-vue, @vitejs/plugin-vue-jsx, ...)
npm error
npm error Could not resolve dependency:
npm error dev @bg-dev/nuxt-directus@"*" from the root project
npm error
npm error Conflicting peer dependency: vue@2.6.14
npm error node_modules/vue
npm error   peer vue@">= 2.5 < 2.7" from @vue/composition-api@1.7.2
npm error   node_modules/@vue/composition-api
npm error     peerOptional @vue/composition-api@"^1.0.0" from @vue/apollo-composable@4.2.1
npm error     node_modules/@vue/apollo-composable
npm error       peer @vue/apollo-composable@"^4.0.1" from @bg-dev/nuxt-directus@2.5.0
npm error       node_modules/@bg-dev/nuxt-directus
npm error         dev @bg-dev/nuxt-directus@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.

**Comment by becem-gharbi** [comment]
The issue is related to the `localhost` dns resolution by Nodejs. In order to run a server-side fetch call from Nuxt to the local Directus server, Nodejs should resolve the `localhost` to `127.0.0.1` and not `::1`

**Comment by becem-gharbi** [comment]
The issue is related to the `localhost` resolution by Nodejs. 

In order to run a server-side fetch call from Nuxt to the local Directus server, Nodejs should resolve the `localhost` to `127.0.0.1` and not `::1` meaning preferring ipv4 over ipv6. This is done by adding `NODE_OPTIONS=--dns-result-order=ipv4first` env variable as mentioned in README. On Nodejs v22 I just found that this didn't solve the issue for some reason. I suggest downgrading to Node v20 if possible.

This is the log error returned when fetching user endpoint on SSR `[cause]: connect ECONNREFUSED ::1:8055`.

**Comment by ghost** [comment]
@becem-gharbi  Thanks I downgraded to v20, and considering I'm using the latest version of Nuxt3 v3.15.0 and my directus served on my VPS by docker also, means I'm not using directus on the localhost

**Comment by becem-gharbi** [comment]
Is your Directus server and Nuxt app have the same domain name? If no then cookies are not lost. Please refer to README.

**Comment by ghost** [comment]
@becem-gharbi the directus on my subdomain served, i'm on localhost for nuxt app (front end), i served the backend (directus) and for frontend, I'm on developing on localhost and when i done that i have to deploy on the main domain on my vps server

**Comment by becem-gharbi** [comment]
Hi again,
On SSR, both the Directus server and Nuxt app should be on the same domain name to send cookies  and perform token refresh and thus subsequent requests such as fetching user data.

Otherwise in your case, having your Nuxt app on localhost and the Directus server on another means cookies should be set to cross site (`samesite`="none"). This requires them to be secured (`secure`="true"). So they cannot be sent to a non secure server (your Nuxt app).
